src/presentation/ui/component/form/fileUploadTextInputFileContentReader.templ

```diff
@@ -0,0 +1,65 @@
+package componentForm
+
+script FileUploadTextInputFileContentReaderLocalState() {
+	document.addEventListener('alpine:init', () => {
+		Alpine.data('fileUploadTextInputFileContentReader', () => ({
+			uploadedFileName: '',
+			get uploadedFileNameLabel() {
+				if (this.uploadedFileName == '') {
+					return 'No file chosen';
+				}
+
+				return this.uploadedFileName;
+			},
+			init() {
+				this.uploadedFileName = '';
+			},
+			handleFileUpload(event) {
+				const inputFiles = Array.from(event.target.files);
+				if (inputFiles.length == 0) {
+					return;
+				}
+
+				const uploadedFile = inputFiles[0];
+				this.uploadedFileName = uploadedFile.name;
+
+				const reader = new FileReader();
+				reader.onload = (event) => {
+					this.$dispatch('file-content-readed', event.target.result);
+				};
+				reader.readAsText(uploadedFile);
+			}
+		}));
+	});
+}
+
+templ FileUploadTextInputFileContentReader(
```

Comment:
You didn't rename this to TextInputFileContentUploadReader.

Reply:
We had agreed that either of the [two proposed names](https://github.com/speedianet/os/pull/171#discussion_r1767238161) would work. I revisited them and found `FileUploadTextInputFileContentReader` more appropriate, as it clearly indicates that it’s a file upload (`FileUpload`) which is transformed into a text input (`FileInput`), where its content is the file content read by a reader (`FileContentReader`).